core-bundle/src/Event/MemberActivationMailEvent.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Event;
+
+use Contao\CoreBundle\OptIn\OptInToken;
+use Contao\MemberModel;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class MemberActivationMailEvent extends Event
+{
+    private bool $sendOptInToken = true;
+
+    public function __construct(private MemberModel $member, private OptInToken $optInToken, private string $subject, private string $text, private array $simpleTokens)
+    {
+    }
+
+    public function getMember(): MemberModel
+    {
+        return $this->member;
+    }
+
+    public function getOptInToken(): OptInToken
+    {
+        return $this->optInToken;
+    }
+
+    public function setSubject(string $subject): self
+    {
+        $this->subject = $subject;
+
+        return $this;
+    }
+
+    public function getSubject(): string
+    {
+        return $this->subject;
+    }
+
+    public function setText(string $text): self
+    {
+        $this->text = $text;
+
+        return $this;
+    }
+
+    public function getText(): string
+    {
+        return $this->text;
+    }
+
+    public function setSimpleTokens(array $simpleTokens): self
+    {
+        $this->simpleTokens = $simpleTokens;
+
+        return $this;
+    }
+
+    public function setSimpleToken(string $token, string $value): self
+    {
+        $this->simpleTokens[$token] = $value;
+
+        return $this;
+    }
+
+    public function getSimpleTokens(): array
+    {
+        return $this->simpleTokens;
+    }
+
+    public function setSendOptInToken(bool $sendOptInToken): self
```

Comment:
Can we not use setters and getters here but be more speaking?

* disableSendingTheOptInToken(): self
* enableSendingTheOptInToken(): self
* shouldSendOptInToken(): bool